components/d2l-filter-dropdown/d2l-filter-dropdown.js

```diff
@@ -138,6 +138,17 @@ class D2LFilterDropdown extends mixinBehaviors([D2L.PolymerBehaviors.FilterDropd
 		}
 	}
 
+	updateFilterOptionSelectedStatus(categoryKey, optionKey, value) {
+		const cIndex = this._getFilterIndexFromKey(categoryKey);
+		if (cIndex >= 0) {
+			const oIndex = this._getOptionIndexFromKey(cIndex, optionKey);
+			if (oIndex >= 0 && this._filters[cIndex].options[oIndex].selected !== value) {
```

Comment:
I assume that the checks you've done ensure that `selected` and `numSelected` on the lower line can never be null?

If that's true, then cool :)